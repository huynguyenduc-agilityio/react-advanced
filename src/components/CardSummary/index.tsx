import { memo, ReactNode } from 'react';
import { VscKebabVertical } from 'react-icons/vsc';
import { Box, Flex, Icon, Text } from '@chakra-ui/react';

export interface CardSummaryProps {
  title: string;
  total: number;
  icon?: ReactNode;
}

const CardSummary = ({ title, total, icon }: CardSummaryProps) => {
  return (
    <Box
      w="full"
      bg="midNightBlue"
      p={5}
      border="0.6px solid"
      borderColor="slateBlue"
      borderRadius="lg"
    >
      <Flex gap={3} alignItems="center">
        {icon && <Box>{icon}</Box>}
        <Flex w="full" alignItems="center" justifyContent="space-between">
          <Box>
            <Text fontSize="lg" fontWeight="medium" color="white">
              {title}
            </Text>
            <Text fontSize="sm" color="pastelBlue">
              {total}
            </Text>
          </Box>
          <Icon as={VscKebabVertical} color="white" boxSize="20px" />
        </Flex>
      </Flex>
    </Box>
  );
};

export default memo(CardSummary);
