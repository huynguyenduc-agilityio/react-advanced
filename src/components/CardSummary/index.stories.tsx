import { Box, Flex } from '@chakra-ui/react';
import type { Meta, StoryObj } from '@storybook/react';

// Icons
import { HeartIcon, KebabIcon, UserGroupIcon, UserIcon } from '@/components';

// Components
import CardSummary from '.';

const meta = {
  title: 'Components/CardSummary',
  component: CardSummary,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <Flex justifyContent="center" w="1000px" p={9} bgColor="spaceBlue">
      <Box w="243px">
        <CardSummary {...args} />
      </Box>
    </Flex>
  ),
} satisfies Meta<typeof CardSummary>;

export default meta;

type Story = StoryObj<typeof CardSummary>;

export const TotalUsers: Story = {
  args: {
    title: 'Total Users',
    total: 250,
    icon: <UserGroupIcon />,
  },
};

export const NewUsers: Story = {
  args: {
    title: 'New Users',
    total: 250,
    icon: <UserIcon />,
  },
};

export const TopUsers: Story = {
  args: {
    title: 'Top Users',
    total: 250,
    icon: <HeartIcon />,
  },
};

export const OtherUsers: Story = {
  args: {
    title: 'Other Users',
    total: 250,
    icon: <KebabIcon />,
  },
};
