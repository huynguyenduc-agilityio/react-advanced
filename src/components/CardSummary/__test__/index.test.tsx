import { render } from '@testing-library/react';

// Components
import CardSummary from '..';

describe('CardSummary component', () => {
  it('renders the Button with children', () => {
    const { getByText } = render(
      <CardSummary title="Top Users" total={250} icon={<span>icon</span>} />,
    );
    expect(getByText('Top Users')).toBeInTheDocument();
    expect(getByText('250')).toBeInTheDocument();
    expect(getByText('icon')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <CardSummary title="Top Users" total={250} icon={<span>icon</span>} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
