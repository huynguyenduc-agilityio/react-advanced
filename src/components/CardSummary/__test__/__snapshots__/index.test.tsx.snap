// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSummary component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-si9sfu"
  >
    <div
      class="css-1plht7o"
    >
      <div
        class="css-0"
      >
        <span>
          icon
        </span>
      </div>
      <div
        class="css-hlrfxo"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-ampera"
          >
            Top Users
          </p>
          <p
            class="chakra-text css-1kfsxnu"
          >
            250
          </p>
        </div>
        <svg
          class="chakra-icon css-1rgvbqi"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.444 13.832a1 1 0 1 0 1.111-1.663 1 1 0 0 0-1.11 1.662zM8 9a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0-5a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
